fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java

```diff
@@ -170,18 +172,20 @@ private static boolean isLockFree(boolean isOnlyOlapTable, ConnectContext sessio
                 && !session.getSessionVariable().isCboUseDBLock();
     }
 
-    private static ExecPlan createQueryPlan(Relation relation,
+    private static ExecPlan createQueryPlan(StatementBase stmt,
                                             ConnectContext session,
                                             TResultSinkType resultSinkType) {
-        QueryRelation query = (QueryRelation) relation;
+        QueryStatement queryStmt = (QueryStatement) stmt;
+        QueryRelation query = (QueryRelation) queryStmt.getQueryRelation();
         List<String> colNames = query.getColumnOutputNames();
         // 1. Build Logical plan
         ColumnRefFactory columnRefFactory = new ColumnRefFactory();
         LogicalPlan logicalPlan;
+        Map<Operator, ParseNode> optToAstMap = Maps.newHashMap();
```

Comment:
Could we collect this map only when enable text based rewrite? Is it time-consuming when the query is complex?

Reply:
fixed.